### ISSUE TITLE
Crash on Build&View

### ISSUE THREAD

**Issue by bewlay** [texstudio-org/texstudio#1542]
I'm using Texstudio 3.1.1 on Debian Buster. Texstudio sometimes crashes when I Build&View, I'd guess less than 5% of the time. When restarting, the properly compiled document is shown. I wasn't able to reliably reproduce this, unfortunately, but I captured a gdb backtrace:

```
texstudio: strtod_l.c:1496: ____strtod_l_internal: Assertion `numsize == 1 && n < d' failed.

Thread 492 "PDFRenderEngine" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fff8d7fa700 (LWP 8431)]
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00007ffff602e7bb in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6019535 in __GI_abort () at abort.c:79
#2  0x00007ffff601940f in __assert_fail_base
    (fmt=0x7ffff617bee0 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x7ffff61784c5 "numsize == 1 && n < d", file=0x7ffff6178500 "strtod_l.c", line=1496, function=<optimized out>)
    at assert.c:92
#3  0x00007ffff6027102 in __GI___assert_fail
    (assertion=assertion@entry=0x7ffff61784c5 "numsize == 1 && n < d", file=file@entry=0x7ffff6178500 "strtod_l.c", line=line@entry=1496, function=function@entry=0x7ffff617faf0 <__PRETTY_FUNCTION__.12238> "____strtod_l_internal") at assert.c:101
#4  0x00007ffff60387c5 in __GI_____strtod_l_internal (nptr=<optimized out>, endptr=<optimized out>, group=<optimized out>, loc=<optimized out>) at strtod_l.c:1496
#5  0x00007ffff5db9910 in FoFiType1::parse() () at /lib/x86_64-linux-gnu/libpoppler.so.82
#6  0x00007ffff5db9d85 in FoFiType1::getName() () at /lib/x86_64-linux-gnu/libpoppler.so.82
#7  0x00007ffff5e26389 in Gfx8BitFont::Gfx8BitFont(XRef*, char const*, Ref, GooString*, GfxFontType, Ref, Dict*) () at /lib/x86_64-linux-gnu/libpoppler.so.82
#8  0x00007ffff5e28cbf in GfxFont::makeFont(XRef*, char const*, Ref, Dict*) () at /lib/x86_64-linux-gnu/libpoppler.so.82
#9  0x00007ffff5e28ea4 in GfxFontDict::GfxFontDict(XRef*, Ref*, Dict*) () at /lib/x86_64-linux-gnu/libpoppler.so.82
#10 0x00007ffff5e0a3bc in GfxResources::GfxResources(XRef*, Dict*, GfxResources*) () at /lib/x86_64-linux-gnu/libpoppler.so.82
#11 0x00007ffff5e15beb in Gfx::Gfx(PDFDoc*, OutputDev*, int, Dict*, double, double, PDFRectangle const*, PDFRectangle const*, int, bool (*)(void*), void*, XRef*) ()
    at /lib/x86_64-linux-gnu/libpoppler.so.82
#12 0x00007ffff5e6b3ba in Page::createGfx(OutputDev*, double, double, int, bool, bool, int, int, int, int, bool, bool (*)(void*), void*, XRef*) () at /lib/x86_64-linux-gnu/libpoppler.so.82
#13 0x00007ffff5e6b634 in Page::displaySlice(OutputDev*, double, double, int, bool, bool, int, int, int, int, bool, bool (*)(void*), void*, bool (*)(Annot*, void*), void*, bool) ()
    at /lib/x86_64-linux-gnu/libpoppler.so.82
#14 0x00007ffff7c106a9 in Poppler::Page::renderToImage(double, double, int, int, int, int, Poppler::Page::Rotation, void (*)(QImage const&, QVariant const&), bool (*)(QVariant const&), bool (*)(QVariant const&), QVariant const&) const () at /lib/x86_64-linux-gnu/libpoppler-qt5.so.1
#15 0x00007ffff7c108c0 in Poppler::Page::renderToImage(double, double, int, int, int, int, Poppler::Page::Rotation, void (*)(QImage const&, QVariant const&), bool (*)(QVariant const&), QVariant const&) const () at /lib/x86_64-linux-gnu/libpoppler-qt5.so.1
#16 0x00007ffff7c1092b in Poppler::Page::renderToImage(double, double, int, int, int, int, Poppler::Page::Rotation) const () at /lib/x86_64-linux-gnu/libpoppler-qt5.so.1
#17 0x0000555555afc0cd in PDFRenderEngine::run() ()
#18 0x00007ffff6652a67 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#19 0x00007ffff64e0fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#20 0x00007ffff60f04cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

```

**Comment by sunderme** [comment]
that is a poppler crash, see https://gitlab.freedesktop.org/poppler/poppler/-/issues/1050 and #1409
Resolved with poppler 21.03.

**Comment by bewlay** [comment]
Thanks a lot, I will try that.
